Sources/AmazonLocationiOSAuthSDK/AuthHelper.swift

```diff
@@ -1,47 +1,70 @@
-import AWSCore
-import AWSMobileClientXCF
-import AWSLocationXCF
+import Foundation
 
 public class AuthHelper {
 
     public init() {
     }
     
-    public func authenticateWithCognitoUserPool(identityPoolId: String, region: String) -> LocationCredentialsProvider {
-        let regionType = AWSEndpoint.regionTypeByString(regionString: region)
-        let credentialProvider : LocationCredentialsProvider = LocationCredentialsProvider(regionType: regionType!, identityPoolId: identityPoolId)
-        
-        let configuration = AWSServiceConfiguration(
-            region: regionType!,
-            credentialsProvider: credentialProvider.getCognitoProvider()
-        )
-        
-        AWSServiceManager.default().defaultServiceConfiguration = configuration
-        return credentialProvider
+    public func authenticateWithCognitoIdentityPool(identityPoolId: String) async throws -> LocationCredentialsProvider? {
+        let regionType = AmazonLocationEndpoint.toRegionType(identityPoolId: identityPoolId)
+        return try? await authenticateWithCognitoIdentityPool(identityPoolId: identityPoolId, regionType: regionType!)
     }
     
-    public func authenticateWithCognitoUserPool(identityPoolId: String) -> LocationCredentialsProvider {
-        let regionType = AWSEndpoint.toRegionType(identityPoolId: identityPoolId)
-        
-        let credentialProvider : LocationCredentialsProvider = LocationCredentialsProvider(regionType: regionType!, identityPoolId: identityPoolId)
+    public func authenticateWithCognitoIdentityPool(identityPoolId: String, region: String) async throws -> LocationCredentialsProvider? {
+        let regionType = AmazonLocationEndpoint.regionTypeByString(regionString: region)
+        return try? await authenticateWithCognitoIdentityPool(identityPoolId: identityPoolId, regionType: regionType!)
+    }
+    
+    private func authenticateWithCognitoIdentityPool(identityPoolId: String, regionType: AmazonLocationRegionType) async throws -> LocationCredentialsProvider? {
+        let credentialProvider : LocationCredentialsProvider = LocationCredentialsProvider(regionType: regionType, identityPoolId: identityPoolId)
+        do {
+            //if let cognitoCredentialString = KeyChainHelper.get(key: .CognitoCrdentials), let savedCognitoCrdentials = CognitoCredentials.decodeCognitoCredentials(jsonString: cognitoCredentialString), savedCognitoCrdentials.expiryDate! < Date() {
```

Comment:
dont leave commented out code. If you need it later you can always get it from git history

Reply:
Fixed